### PR TITLE
Update Workflow badge

### PR DESCRIPTION
**Pull request by NdagiStanley**
* [x] I have added an entry to `docs/changelog.md`

## Summary of changes

Update the link to the GitHub actions workflow status badge based on the [updated docs](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge).

### Considerations

- I've not used the branch parameter (by appending `?branch=main`) to the link since that shows `no status`:

    

- I've not used the `test_all_packages_slow.yml` workflow file since that is what was originally linked to.

## Test plan

Tested by running the following commands (in a python virtual environment):

```
pip install nox
nox -s watch_docs
```

The resultant home page of the docs, published locally, has a working GitHub actions workflow status badge.

Green arrow points to a working badge.

Screenshot 2021-11-16 at 15 41 43

### CODE REVIEW

**Review by gaborbernat — CHANGES_REQUESTED** [review]
Merge conflicts need to be resolved.

File: src/pipx/commands/reinstall.py [hunk 1]
```diff
@@ -27,7 +27,15 @@ def reinstall(
...
-    if Path(python).is_relative_to(venv_dir):
+    try:
+        # use PurePath.relative_to in a try block instead
+        # of PurePath.is_relative_to, for python 3.6-3.8 compatability
+        Path(python).relative_to(venv_dir)
+        python_relative_to_venv_dir = True
+    except ValueError:
+        python_relative_to_venv_dir = False
+
+    if python_relative_to_venv_dir:
```
Comment (cs01) [inline src/pipx/commands/reinstall.py:38]: This looks like valuable change. Could it be made in a different PR?

**Review by cs01 — CHANGES_REQUESTED** [review]
